core/gui/src/app/workspace/component/workspace.component.ts

```diff
@@ -119,16 +120,28 @@ export class WorkspaceComponent implements AfterViewInit, OnInit, OnDestroy {
   }
 
   ngOnDestroy() {
+    this.saveLatestWorkflow();
     this.workflowActionService.setWorkflowMetadata(undefined);
     this.workflowActionService.destroySharedModel();
     this.workflowWebsocketService.closeWebsocket();
   }
 
   @HostListener("window:beforeunload")
   unloadHandler() {
+    this.saveLatestWorkflow();
     this.vc.clear();
   }
 
+  saveLatestWorkflow() {
+    if (this.workflowPersistService.isWorkflowPersistEnabled()) {
+      const workflow = this.workflowActionService.getWorkflow();
+      this.workflowPersistService
+        .persistWorkflow(workflow)
+        .pipe(takeUntil(this._onSavingWorkflowObservable))
```

Comment:
I don't know why you need a subject here. can you remove it?

Reply:
removed